config-templates/tables.postgres.sql

```diff
@@ -0,0 +1,38 @@
+CREATE TABLE "statistics_sums" (
+  "id" bigint GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+  "year" integer NOT NULL,
+  "month" integer DEFAULT NULL,
+  "day" integer DEFAULT NULL,
+  "idp_id" integer NOT NULL DEFAULT 0,
+  "sp_id" integer NOT NULL DEFAULT 0,
+  "logins" integer DEFAULT NULL,
+  "users" integer DEFAULT NULL,
+  CONSTRAINT "year" UNIQUE ("year","month","day","idp_id","sp_id")
+);
+
+CREATE INDEX "statistics_sums_idp_id" ON "statistics_sums" ("idp_id");
+CREATE INDEX "statistics_sums_sp_id" ON "statistics_sums" ("sp_id");
+
+CREATE TABLE "statistics_per_user" (
+  "day" date NOT NULL,
+  "idp_id" integer NOT NULL,
+  "sp_id" integer NOT NULL,
+  "user" character varying(255) NOT NULL,
+  "logins" integer DEFAULT '1',
+  CONSTRAINT primary_key PRIMARY KEY ("day","idp_id","sp_id","user")
+);
+
+CREATE INDEX "statistics_per_user_idp_id" ON "statistics_per_user" ("idp_id");
+CREATE INDEX "statistics_per_user_sp_id" ON "statistics_per_user" ("sp_id");
+
+CREATE TABLE "statistics_idp" (
+  "idp_id" integer GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
```

Comment:
I am wondering if we should use INT or BIGINT. Probably won't reach the limit of INT, so if others are fine with INT, I am as well :).

Reply:
From [PostgreSQL docs](https://www.postgresql.org/docs/9.1/datatype-numeric.html):

> The type integer is the common choice, as it offers the best balance between range, storage size, and performance.
> The bigint type should only be used if the range of the integer type is insufficient, because the latter is definitely faster.

In the IdP and SP table, you would need 2147483647 (over 2 billion) unique identifiers to run out of range. That would mean that 2 out of 7 people on the planet have a unique SP or IdP...

(there is bigint in the statistics_sums table)

Reply:
Also, this is only a template, a user of this module may change the column type to bigint if needed.